### DISCUSSION TITLE
Please share some examples of contract interaction : )

### DISCUSSION THREAD

**Discussion by raffle-x-dev** [rainbow-me/rainbowkit#1720] [Q&A]
Hi there!
I just made a fresh new installation of RainbowKit through CLI with npm init @rainbow-me/rainbowkit@latest.
I wish to do some basic contract interactions: read and write, calling functions.
I can't find any examples btw, and wagmi seems to have a different syntax that either doesn't work in your framework or throws errors I can't solve.
Can RainbowKit do this out of the box or do I need to install other dependencies?

Thank you!

**Comment by 0xmkh** [comment]
@raffle-x-dev What wagmi and viem version are you using ? If you have viem `~1.21.4` installed with wagmi `~1.4.13` version it should work fine after running `npm init @rainbow-me/rainbowkit@latest`. Here is one of our [example](https://github.com/rainbow-me/rainbowkit/blob/main/examples/with-next-mint-nft/pages/index.tsx#L27-L38) of how we use `wagmi`s contract to mint an NFT.

Let me know if you're facing with any other issues.

**Reply by raffle-x-dev** [reply]
Hey there!
I manage to make it work somehow, but now I'm facing Hydration errors when using `useContractRead`.
These are my dependencies:
`{
  "name": "rk",
  "private": true,
  "version": "0.1.0",
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@rainbow-me/rainbowkit": "^1.3.3",
    "next": "^14.0.4",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.21.3",
    "viem": "~1.21.4",
    "wagmi": "~1.4.13"
  },
  "devDependencies": {
    "@types/node": "^18.19.3",
    "@types/react": "^18.2.43",
    "eslint": "^8.15.0",
    "eslint-config-next": "^14.0.4",
    "typescript": "^5.0.4"
  }
}`

**Reply by DanielSinclair** [reply]
Are you using SSR? If so, ensure that you're calling Wagmi hooks on the frontend. Here is a full example use contract interactions to mint an NFT: [with-next-mint-nft](https://github.com/rainbow-me/rainbowkit/tree/main/examples/with-next-mint-nft). Here is a Next.js App Router example for clarity on a proper structured SSR dApp: [with-next-app](https://github.com/rainbow-me/rainbowkit/tree/main/examples/with-next-app).
